.jenkins/library/vars/helpers.groovy

```diff
@@ -87,14 +87,15 @@ def CmakeArgs(Map args) {
 
 def WaitForAptLock() {
     return """
+        ${needSudo()}
         counter=0
         max=600
         step=5
         echo "Checking for locks..."
-        while sudo fuser /var/lib/dpkg/lock > /dev/null 2>&1 ||
-              sudo fuser /var/lib/dpkg/lock-frontend > /dev/null 2>&1 ||
-              sudo fuser /var/lib/apt/lists/lock > /dev/null 2>&1 ||
-              sudo ps aux | grep -E "[a]pt|[d]pkg"; do
+        while \${maybesudo} fuser /var/lib/dpkg/lock > /dev/null 2>&1 ||
```

Comment:
why maybesudo?

Reply:
This is to allow the the use of the same helper function in cases where sudo is available, as well as when sudo is not available. E.g. OE's base Docker images do not have sudo installed, and run as root. In VM and oetools Docker image, sudo is available and run as non-root with sudoers.